packages/core/src/modules/dids/services/DidService.ts

```diff
@@ -0,0 +1,133 @@
+import type { AgentContext } from '../../../agent'
+import type { Key } from '../../../crypto'
+import type { Logger } from '../../../logger'
+import type { MediationRecord } from '../../routing/repository/MediationRecord'
+import type { DidDocument } from '../domain'
+import type { DidCreateResult } from '../types'
+
+import { AgentConfig } from '../../../agent/AgentConfig'
+import { InjectionSymbols } from '../../../constants'
+import { KeyType } from '../../../crypto'
+import { injectable, inject } from '../../../plugins'
+import { uuid } from '../../../utils/uuid'
+import { Wallet } from '../../../wallet/Wallet'
+import { DidDocumentBuilder } from '../domain'
+import { getEd25519VerificationMethod } from '../domain/key-type/ed25519'
+import { getX25519VerificationMethod } from '../domain/key-type/x25519'
+import { DidCommV2Service } from '../domain/service'
+import { DidRegistrarService } from '../services/DidRegistrarService'
+
+export interface DIDRoutingParams {
+  endpoint?: string
+  mediator?: MediationRecord
+}
+
+export interface CreateDIDParams {
+  seed?: string
+  method?: string
+  routing?: DIDRoutingParams
+}
+
+export interface CreateDIDDocumentParams {
+  authentications: Key[]
+  keyAgreements: Key[]
+  services: DidCommV2Service[]
+}
+
+@injectable()
+export class DidService {
```

Comment:
This is just a temporary service to create Dvd's in DidComm V2-compatible way.  We did it to not accidentally break V1 logic because it is tide related to routing and did-exchange.   In the future, we will drop it and integrate it with DidRegistart which was added in 0.3.